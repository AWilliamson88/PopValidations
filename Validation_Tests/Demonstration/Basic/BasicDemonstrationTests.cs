using ApprovalTests;
using PopValidations_Tests.TestHelpers;
using System.Threading.Tasks;
using Xunit;

namespace PopValidations_Tests.Demonstration.Basic;

public class BasicDemonstrationTests
{
    [Fact]
    public async Task BasicSongValidator_Validate_ToJson()
    {
        // Arrange
        var runner = ValidationRunnerHelper.BasicRunnerSetup(new BasicSongValidator());
        var song = new BasicSong(
            "Disturbed",
            null,
            "Down With The Sickness",
            2.4,
            string.Empty
        );

        // Act
        var results = await runner.Validate(song);
        var json = JsonConverter.ToJson(results);

        // Assert
        Approvals.VerifyJson(json);
    }

    [Fact]
    public void BasicSongValidator_Describe_ToJson()
    {
        // Arrange
        var runner = ValidationRunnerHelper.BasicRunnerSetup(new BasicSongValidator());

        // Act
        var results = runner.Describe();
        var json = JsonConverter.ToJson(results);

        // Assert
        Approvals.VerifyJson(json);
    }

    [Fact]
    public async Task BasicArtistValidator_Validate_ToJson()
    {
        // Arrange
        var runner = ValidationRunnerHelper.BasicRunnerSetup(new BasicArtistValidator());
        var artist = new BasicArtist(
            "Disturbed",
            null
        );

        // Act
        var results = await runner.Validate(artist);
        var json = JsonConverter.ToJson(results);

        // Assert
        Approvals.VerifyJson(json);
    }

    [Fact]
    public void BasicArtistValidator_Describe_ToJson()
    {
        // Arrange
        var runner = ValidationRunnerHelper.BasicRunnerSetup(new BasicArtistValidator());

        // Act
        var results = runner.Describe();
        var json = JsonConverter.ToJson(results);

        // Assert
        Approvals.VerifyJson(json);
    }
}

