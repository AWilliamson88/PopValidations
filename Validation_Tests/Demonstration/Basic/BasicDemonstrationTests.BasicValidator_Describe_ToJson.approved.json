{
    "results": [
        {
            "property": "TrackNumber",
            "outcomes": [
                {
                    "validator": "NotNullValidation",
                    "message": "Must not be null.",
                    "values": []
                }
            ],
            "validationGroups": []
        },
        {
            "property": "TrackName",
            "outcomes": [
                {
                    "validator": "IsEqualToValidation",
                    "message": "Must equal to 'Definitely Not The Correct Song Name.'",
                    "values": [
                        {
                            "key": "value",
                            "value": "Definitely Not The Correct Song Name."
                        }
                    ]
                }
            ],
            "validationGroups": []
        },
        {
            "property": "Duration",
            "outcomes": [
                {
                    "validator": "IsEqualToValidation",
                    "message": "Songs must force you to travel slowly backwards in time to listen to.",
                    "values": [
                        {
                            "key": "value",
                            "value": "-1"
                        }
                    ]
                }
            ],
            "validationGroups": []
        }
    ]
}